pkg/kn/flags/bool.go

```diff
@@ -0,0 +1,104 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flags
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+var negPrefix = "no-"
+
+// AddBothBoolFlags adds the given flag in both `--foo` and `--no-foo` variants.
+// If you do this, make sure you call ReconcileBoolFlags later to catch errors and
+// set the relationship between the flag values.
+func AddBothBoolFlags(f *pflag.FlagSet, p *bool, name, short string, value bool, usage string) {
+
+	negativeName := negPrefix + name
+
+	f.BoolVarP(p, name, short, value, usage)
+	f.Bool(negativeName, !value, "do not "+usage)
+
+	if value {
```

Comment:
Just want make sure I grok this... if the default value is `true` then we'll show the non-no flag. Isn't that backwards? Shouldn't we show the `--no-xxx` flag instead since the more obvious option is to turn it off?  Or an I reading this backwards?

Reply:
Take example of `--force` flag:
default value is `false`, we don't show `--no-force` because thats implied, we instead show `--force` so user choose if they want to change the behavior by flagging `--force`.


Another example of `--headers` flag for listing:
default value true, we don't show what is obvious and default behavior, 
but show `--no-headers` to change the default behavior.

Reply:
ok never mind - I was reading it backwards :-)